### DISCUSSION TITLE
Communication for first installation on sonoff mini r4 extreme

### DISCUSSION THREAD

**Discussion by Ripper346** [arendst/Tasmota#21707] [Q&A] [answered]
Hi, I have a Sonoff mini r4 extreme and wanted to install Tasmota. I have found several guides, but I am not able to communicate with the switch. So, what have I done so far:
- Soldered the 4 pins
- Connected to the serial board (to make sure it was a good one, I purchased the same model from a guide) while pressing the button on the switch
- The switch doesn't blink, it doesn't do anything (not sure if it's the expected behavior)
- With tasmota web interface, it says it has failed to communicate
- Same with esptool, I am not even able to get the flash_id, but when using it, I see a led blinking in the serial board, so it is doing something.
- If I don't press the button on the switch when power on, it will blink blue
- I have tried both to get the 3.3V from the serial and an external source, but nothing has changed

Am I missing something? I don't know, for example, if I have to press the button for x seconds. Thank you

Here are some photos if needed

![photo_2024-06-29_18-14-50](https://github.com/arendst/Tasmota/assets/8243693/c78660cc-c69c-42ba-9713-286b93ba2299)

![photo_2024-06-29_18-14-41](https://github.com/arendst/Tasmota/assets/8243693/762c2bf3-7d7e-4170-8e5c-f658e01df90a)

**Comment by sfromis** [comment, marked as answer]
If you got that USB/serial adapter based on a guide, you used a dubious guide, instead of the one from Tasmota, where a better adapter (with more current available) is recommended.
https://tasmota.github.io/docs/Getting-Started/

But if you have a good 3.3V power source on the side, that should make it work better, as long as you also make sure to have common ground for all 3 things, also if you do not get the power from the adapter.

Booting correctly, pushing the button before/while applying power (like a couple of seconds), the expected result should be no led activity, like you see.

And it looks like your wiring is bad, you should connect tx on one end to rx on the other, and vice versa.

**Comment by lalo-uy** [comment]
Tx and rx definition are allways dificult to be sure.
Traditionally tx pin in a modem was the input pin for data to be
“transmited” by the modem.
So when you hookup allway try tx->rx  rx->tx and if fail tx-> tx rx-> rx.
;-)

**Comment by GustavoIO** [comment]
Did you figure it out? I'm having the same issue. I've used this adapter for many sonoff before, but never mini r4...

**Reply by sfromis** [reply]
The weak adapter shown in the image has better chances of working with legacy ESP8266 than with the current-generation ESP32 used here. Not recommended. There are reasons for us recommending another cheap adapter (see the link already posted), able to provide 150 mA of 3.3V power. Should be fine for flashing, but do power with mains instead to get on Wifi.

**Reply by Ripper346** [reply]
I can confirm, I used a nodemcu with esp32 and for another I got the adapter they suggest. The one in the photo didn't ever worked with them

**Reply by GustavoIO** [reply]
After some trial and error I got it to work. It seems I had to set the baudrate to 1152000, I tried 9600 initially to make it slower, but that's too slow. 115200 did the trick.
